dashboard/app/controllers/user_levels_controller.rb

```diff
@@ -23,9 +23,27 @@ def destroy
     head :no_content
   end
 
+  def delete_predict_level_progress
+    script = Script.get_from_cache(params[:script_id])
+    return head :not_found, text: 'Script not found' unless script
+    return head :forbidden, text: 'User must be instructor of course' unless script.can_be_instructor?(current_user)
+    level = Level.find(params[:level_id])
+    return head :not_found, text: 'Level not found' unless level
+    return head :bad_request, text: "Clearing progress on level type #{level.type} is not supported" unless ['Multi', 'FreeResponse'].include?(level.type)
+    UserLevel.where(user_id: current_user.id, script_id: script.id, level: level.id).destroy_all
+    return head :ok
+  end
+
+  # GET /user_levels/get_token
+  def get_token
```

Comment:
How will this be used?

Reply:
High level, it'll be used to fetch the CSRF token on level pages (which are cached) before the call to delete progress. More detail in [this thread](https://codedotorg.slack.com/archives/CA3KCSGTD/p1660861615792319).

Reply:
Ah, I missed that! Sounds great 👍